src/test/java/sysc4806/project24/mini_survey_monkey/BananaMockTest.java

```diff
@@ -7,15 +7,24 @@
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
 import sysc4806.project24.mini_survey_monkey.controllers.BananaController;
+import sysc4806.project24.mini_survey_monkey.interceptors.SurveyStateInterceptor;
+import sysc4806.project24.mini_survey_monkey.repositories.SurveyRepository;
 
 @WebMvcTest(BananaController.class)
 public class BananaMockTest {
 
     @Autowired
     private MockMvc mockMvc;
 
+    // This prevents SurveyStateInterceptor from being loaded
+    @MockBean
+    private SurveyStateInterceptor surveyStateInterceptor;
```

Comment:
@BraedenKloke This seemed to fix the test failures. SurveyStateInterceptor has a SurveyRepository dependency which was failing to load. `@MockBean` mocks the whole dependency preventing any of its dependencies from loading.

I'm not sure if this is the best solution, but it works!

Reply:
Also, the interceptors have other issues that need to be fixed which I will create another PR for.

Reply:
Fixed in https://github.com/JackieSL1/mini-survey-monkey/pull/71